### PR TITLE
Fix BLAKE2s reporting the same EVP_MD_get_size() as BLAKE2b (64)

### PR DESCRIPTION
**Pull request by nabijaczleweli**
Fixes: commit 6d1e730a1ea2c64bdffa88c6b3bee4c3f5bed602 ("Implement BLAKE2s with the same macro as BLAKE2b")
Closes: #22708

### CODE REVIEW

**Review by paulidale — APPROVED** [review]
Nice job figuring this out so quickly.

**Comment by paulidale** [comment]
Could the failure case be added to `test/recipes/30-test_evp_data/evpmd_blake.txt` please?

**Comment by nabijaczleweli** [comment]
The original reproducer happens in a HKDF and that file is just hashes (MD in openssl parlance(?)), so idk if that's the right place for it (and if it is, no clue how i'd add it in there – 
```
$ find -name '*hkdf*'
./test/recipes/30-test_evp_data/evppkey_kdf_hkdf.txt
./test/recipes/30-test_evp_data/evpkdf_hkdf.txt
```
but neither of these files mention BLAKE, and both mention they're from RFC test vectors).

The actual issue was that `EVP_MD_get_size(blake2s_md);` was 64 instead of 32, so I would maybe add a test for that, but then again no clue where.

**Comment by paulidale** [comment]
Just checking the `EVP_MD_get_size(blake2s_md)` could be done in either `test/evp_extra_test.c` or `test/evp_extra_test2.c` and that would be sufficient.

Alternatively, adding something to `test/recipes/30-test_evp_data/evpkdf_hkdf.txt` is probably easier.  Using a new digest here isn't a problem IMO.

**Comment by nabijaczleweli** [comment]
Test added, works with the first patch and fails without.

**Review by paulidale — APPROVED** [review]

**Review by t-j-h — APPROVED** [review]

**Comment by t8m** [comment]
CI failure is relevant.

File: test/recipes/30-test_evp_data/evpkdf_hkdf.txt [hunk 1]
```diff
@@ -213,3 +213,10 @@ Ctrl.info = hexinfo:c1c2c3
 Ctrl.info = hexinfo:c4c5c6c7c8c9cacbcccdcecfd0d1d2d3d4d5d6d7d8d9
 Ctrl.info = hexinfo:dadbdcdddedfe0e1e2e3e4e5e6e7e8e9eaebecedeeeff0f1f2f3f4f5f6f7f8f9fafbfcfdfeff
 Output = 0bd770a74d1160f7c9f12cd5912a06ebff6adcae899d92191fe4305673ba2ffe8fa3f1a4e5ad79f3f334b3b202b2173c486ea37ce3d397ed034c7f9dfeb15c5e927336d0441f4c4300e2cff0d0900b52d3b4
+
+KDF = HKDF
```
Comment (t8m) [inline test/recipes/30-test_evp_data/evpkdf_hkdf.txt:217]: Adding `Availablein = default` should fix the failure. However I am also curious if this works for builds with configure option `no-blake2`.

**Review by t8m — CHANGES_REQUESTED** [review]

**Comment by t-j-h** [comment]
I've dropped my approval - the CI issues do need checking and resolving first.

File: test/recipes/30-test_evp_data/evpkdf_hkdf.txt (hunk 1 above)
Comment (nabijaczleweli) [reply test/recipes/30-test_evp_data/evpkdf_hkdf.txt:217]: Doesn't appear to, `./Configure no-blake2` appears to break more tests than just this
```
Test Summary Report
-------------------
01-test_sanity.t                      (Wstat: 256 (exited 1) Tests: 1 Failed: 1)
  Failed test:  1
  Non-zero exit status: 1
01-test_symbol_presence.t             (Wstat: 512 (exited 2) Tests: 3 Failed: 2)
  Failed tests:  2-3
  Non-zero exit status: 2
01-test_test.t                        (Wstat: 256 (exited 1) Tests: 1 Failed: 1)
  Failed test:  1
  Non-zero exit status: 1
02-test_errstr.t                      (Wstat: 34816 (exited 136) Tests: 137 Failed: 136)
  Failed tests:  1-136
  Non-zero exit status: 136
02-test_lhash.t                       (Wstat: 256 (exited 1) Tests: 1 Failed: 1)
  Failed test:  1
  Non-zero exit status: 1
02-test_list.t                        (Wstat: 256 (exited 1) Tests: 1 Failed: 1)
  Failed test:  1
  Non-zero exit status: 1
02-test_localetest.t                  (Wstat: 768 (exited 3) Tests: 3 Failed: 3)
  Failed tests:  1-3
  Non-zero exit status: 3
02-test_priority_queue.t              (Wstat: 256 (exited 1) Tests: 1 Failed: 1)
  Failed test:  1
  Non-zero exit status: 1
02-test_stack.t                       (Wstat: 256 (exited 1) Tests: 1 Failed: 1)
  Failed test:  1
  Non-zero exit status: 1
03-test_exdata.t                      (Wstat: 256 (exited 1) Tests: 1 Failed: 1)
  Failed test:  1
  Non-zero exit status: 1
03-test_internal_ssl_cert_table.t     (Wstat: 256 (exited 1) Tests: 1 Failed: 1)
  Failed test:  1
  Non-zero exit status: 1
03-test_params_api.t                  (Wstat: 256 (exited 1) Tests: 1 Failed: 1)
  Failed test:  1
  Non-zero exit status: 1
03-test_property.t                    (Wstat: 256 (exited 1) Tests: 2 Failed: 1)
  Failed test:  2
  Non-zero exit status: 1
03-test_ui.t                          (Wstat: 256 (exited 1) Tests: 1 Failed: 1)
  Failed test:  1
  Non-zero exit status: 1
04-test_asn1_decode.t                 (Wstat: 256 (exited 1) Tests: 1 Failed: 1)
  Failed test:  1
  Non-zero exit status: 1
04-test_asn1_encode.t                 (Wstat: 256 (exited 1) Tests: 1 Failed: 1)
  Failed test:  1
  Non-zero exit status: 1
04-test_asn1_encode.t                 (Wstat: 256 (exited 1) Tests: 1 Failed: 1)
  Failed test:  1
  Non-zero exit status: 1
04-test_asn1_string_table.t           (Wstat: 256 (exited 1) Tests: 1 Failed: 1)
  Failed test:  1
  Non-zero exit status: 1
04-test_bio_callback.t                (Wstat: 256 (exited 1) Tests: 1 Failed: 1)
  Failed test:  1
  Non-zero exit status: 1
04-test_bio_core.t                    (Wstat: 256 (exited 1) Tests: 1 Failed: 1)
  Failed test:  1
  Non-zero exit status: 1
04-test_bio_dgram.t                   (Wstat: 256 (exited 1) Tests: 1 Failed: 1)
  Failed test:  1
  Non-zero exit status: 1
04-test_bioprint.t                    (Wstat: 256 (exited 1) Tests: 1 Failed: 1)
  Failed test:  1
  Non-zero exit status: 1
04-test_conf.t                        (Wstat: 512 (exited 2) Tests: 4 Failed: 2)
  Failed tests:  1, 3
  Non-zero exit status: 2
04-test_err.t                         (Wstat: 256 (exited 1) Tests: 1 Failed: 1)
  Failed test:  1
  Non-zero exit status: 1
04-test_membio.t                      (Wstat: 256 (exited 1) Tests: 1 Failed: 1)
  Failed test:  1
  Non-zero exit status: 1
04-test_params_conversion.t           (Wstat: 256 (exited 1) Tests: 1 Failed: 1)
  Failed test:  1
  Non-zero exit status: 1
04-test_pem_read_depr.t               (Wstat: 256 (exited 1) Tests: 1 Failed: 1)
  Failed test:  1
  Non-zero exit status: 1
04-test_pem_reading.t                 (Wstat: 5376 (exited 21) Tests: 55 Failed: 21)
  Failed tests:  1-3, 5-9, 11-14, 16, 18, 21-27
  Non-zero exit status: 21
30-test_evp.t                         (Wstat: 256 (exited 1) Tests: 79 Failed: 1)
  Failed test:  7
  Non-zero exit status: 1
Files=295, Tests=3605, 706 wallclock secs (11.06 usr  0.83 sys + 616.98 cusr 87.12 csys = 715.99 CPU)
Result: FAIL
```

**Comment by nabijaczleweli** [comment]
`Availablein = default` seems to've fixed the CI.

File: test/recipes/30-test_evp_data/evpkdf_hkdf.txt (hunk 1 above)
Comment (t8m) [reply test/recipes/30-test_evp_data/evpkdf_hkdf.txt]: This must have been some glitch when making the library or something.

**Review by paulidale — APPROVED** [review]

**Comment by t8m** [comment]
Merged to the master branch. Thank you for your contribution.

**Pull request closed without merging**